data/src/main/java/com/linkedin/data/schema/validation/ValidateDataAgainstSchema.java

```diff
@@ -278,13 +292,82 @@ protected void recurseRecord(DataElement element, RecordDataSchema schema, DataM
       {
         String key = entry.getKey();
         RecordDataSchema.Field field = schema.getField(key);
+        Object value = entry.getValue();
         if (field != null)
         {
-          Object value = entry.getValue();
           DataSchema childSchema = field.getType();
           childElement.setValueNameSchema(value, key, childSchema);
           validate(childElement, childSchema, value);
         }
+        else
+        {
+          childElement.setValueNameSchema(value, key, null);
+          handleUnrecognizedField(childElement);
```

Comment:
Also what about checking the unrecognized field mode at this point in time like we do for iteratively?